core/src/main/scala/chisel3/experimental/HasTypeAlias.scala

```diff
@@ -0,0 +1,51 @@
+package chisel3.experimental
+
+import chisel3.Record
+
+/** Wrapper object for a Record alias name. Primarily intended to provide an invocation point for source line locators, but
+  * also contains pertinent information to generating FIRRTL alias statements.
+  *
+  * @param id The desired name to generate an alias statement
+  * @param strippedSuffix In the case of forced coersion by [[Input]] or [[Output]], the string to append to the end of the
+  *        alias name. Takes the default value of `"_stripped"`
+  */
+case class BundleAlias private[chisel3] (info: SourceInfo, id: String, strippedSuffix: String = "_stripped")
+
+object BundleAlias {
+  def apply(id: String)(implicit info:  SourceInfo): BundleAlias = BundleAlias(info, id)
+  def apply(id: String, strippedSuffix: String)(implicit info: SourceInfo): BundleAlias =
+    BundleAlias(info, id, strippedSuffix)
+}
+
+trait HasTypeAlias {
```

Comment:
should this be called MightHaveTypeAlias? Since it seems like it might still not have one at all?